src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1629,6 +1630,45 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
 
             lock (element3DDictionaryMutex)
             {
+                // Instancing
+                var pkgs = packages.ToList();
+                if (pkgs.Count > 1)
+                {
+                    var p = pkgs[0];
+                    if (p.Mesh != null && p.Mesh.Positions.Any())
+                    {
+                        var baseId = p.Description;
+                        if (baseId.IndexOf(":", StringComparison.Ordinal) > 0)
+                        {
+                            baseId = baseId.Split(':')[0];
+                        }
+                        var id = ((p.RequiresPerVertexColoration || p.Colors != null || p.RequiresCustomTransform)
+                            ? p.Description : baseId) + MeshKey;
+
+                        InstancingMeshGeometryModel3D instancingModel;
+                        if (Element3DDictionary.TryGetValue(id, out var element3D))
+                        {
+                            instancingModel = element3D as InstancingMeshGeometryModel3D;
+                        }
+                        else
+                        {
+                            instancingModel =  new InstancingMeshGeometryModel3D();
```

Comment:
hmm - we should inspect the EffectsManager at runtime to see if theres an effect for instancing meshes - it may try to use our shader.